tests/LocalizatorTest.php

```diff
@@ -165,4 +165,44 @@ public function testLocalizeCommandWhereKeysAreEscapedWithSlashes(): void
         // Cleanup.
         self::flushDirectories('lang', 'views');
     }
+
+    public function testLocalizeCommandReturnsNonZeroCodeWhenCiFlagIsOnAndSomeMissingTranslationsHaveBeenAdded(): void
+    {
+        $this->createTestView("{{ __('Hi mom') }}");
+
+        // Run localize command.
+        $this->artisan('localize', ['--ci' => true])
+            ->assertExitCode(1);
+
+        // Cleanup.
+        self::flushDirectories('lang', 'views');
+    }
+
+    public function testLocalizeCommandReturnsZeroCodeWhenCiFlagIsOnAndNoMissingTranslationsHaveBeenFound(): void
+    {
+        $this->createTestView("");
+
+        // Run localize command.
+        $this->artisan('localize', ['--ci' => true])
+            ->assertExitCode(0);
+
+        // Cleanup.
+        self::flushDirectories('lang', 'views');
+    }
+
+    public function testLocalizeCommandReturnsNonZeroCodeThenZeroCodeIfFileDidNotChangedBetweenTwoRun(): void
```

Comment:
This one will not pass, because I feel I will need something new that seems to not exist yet. Let me explain:

For this test to pass, the parser should detect, among the keys it has collected, which ones are **not** yet in all translated files. This seems like this piece of logic mixes both the Parser and the Writers, since the Writers are aware of which translations files contains which keys, when the Parser only knows which keys are in the "app" files.

My first suggestion would be to have a method `Parser::foundSomeNewKeysComparedTo(Collection $keys): bool`.

What do you think about it @amiranagram?

Reply:
Good job overall!

As for this you can use `Amirami\Localizator\Services\Localizator` class to merge the keys together and figure out if there are any. Look at the `collect` method there, it does everything. You can make it public.

So like:

```php
$localizator->collect($parser->getKeys($locale, $type), $type, $locale);
```

Obviously need to loop over all types and locales.